Corona-Warn-App/src/main/java/de/rki/coronawarnapp/worker/DiagnosisTestResultRetrievalPeriodicWorker.kt

```diff
@@ -73,35 +71,57 @@ class DiagnosisTestResultRetrievalPeriodicWorker @AssistedInject constructor(
         return result
     }
 
+    private fun abortConditionsMet(): Boolean {
+        if (LocalData.isTestResultAvailableNotificationSent()) {
+            Timber.tag(TAG).d("$id: Notification already sent.")
+            return true
+        }
+        if (submissionSettings.hasViewedTestResult.value) {
+            Timber.tag(TAG).d("$id: Test result has already been viewed.")
+            return true
+        }
+
+        if (TimeAndDateExtensions.calculateDays(
+                LocalData.initialPollingForTestResultTimeStamp(),
+                System.currentTimeMillis()
+            ) >= BackgroundConstants.POLLING_VALIDITY_MAX_DAYS
+        ) {
+            Timber.tag(TAG)
+                .d(" $id Maximum days of ${BackgroundConstants.POLLING_VALIDITY_MAX_DAYS} days for polling exceeded.")
+            return true
+        }
+
+        return false
+    }
+
     /**
      * Notification Initiation
      *
      * If the returned Test Result is Negative, Positive or Invalid
      * The Background polling  will be stopped
      * and a notification is shown, but only if the App is not in foreground
      *
-     * @see LocalData.isTestResultNotificationSent
+     * @see LocalData.isTestResultAvailableNotificationSent
      * @see LocalData.initialPollingForTestResultTimeStamp
      * @see TestResult
      */
-    private suspend fun initiateNotification(testResult: TestResult) {
-        if (LocalData.isTestResultNotificationSent() || LocalData.submissionWasSuccessful()) {
-            Timber.tag(TAG).d("$id: Notification already sent or there was a successful submission")
-            return
-        }
-        Timber.tag(TAG).d("$id: Test Result retried is $testResult")
+    private suspend fun initiateTestResultAvailableNotification(testResult: TestResult) {
```

Comment:
Maybe we could have this return a  `boolean` and then pull the `stopWorker()` up, so all `stopWorker()` calls are part of the top level function.

Reply:
I moved some logic so that the stopWorker is on top level